base/promotion.jl

```diff
@@ -7,27 +7,23 @@ typejoin(t::ANY) = (@_pure_meta; t)
 typejoin(t::ANY, ts...) = (@_pure_meta; typejoin(t, typejoin(ts...)))
 function typejoin(a::ANY, b::ANY)
     @_pure_meta
-    if isa(a,TypeConstructor); a = a.body; end
-    if isa(b,TypeConstructor); b = b.body; end
     if a <: b
```

Comment:
Yes, it shouldn't be legal, but it turned out to be very hard to eliminate all cases where we do that. It will typically return `false` for such calls.

This also happens when you manually construct a type with free type vars, e.g. `Array{TypeVar(...)}` and pass it to a function. Dispatch uses `Type{Array{TypeVar(...)}}`. I briefly tried putting a stop to that, but it will have to be a project for another day.